### ISSUE TITLE
toJSON() repeats pending updates to nested keys

### ISSUE THREAD

**Issue by mstniy** [parse-community/Parse-SDK-JS#1452]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

If there are pending updates to nested fields, the object returned by `toJSON()` is corrupted.

### Steps to reproduce

```
a = new Parse.Object("MyObject");
a.set('obj', {});
await a.save();
a.set('obj.a', 0);
console.log(a.toJSON());
```

### Actual Outcome

Prints an object with the keys `{obj: {a: 0}, 'obj.a': 0}`

### Expected Outcome

Should print roughly `{obj: {a: 0}}`
###  Environment

Server
- Parse Server version: `irrelevant`
- Operating system: `Ubuntu 20.04`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `5.0.6`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `local`

Client
- Parse JS SDK version: `alpha (d084ac9be058d9b13f8ac2bbf04615f76dc2800d)`

**Comment by mtrezza** [comment]
Did you mix expected and actual outcome?

**Comment by mstniy** [comment]
Whoops. Fixed.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-alpha.8](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0-alpha.8)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [5.1.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/5.1.0)
